### PR TITLE
Initial test-fixing work

### PR DESCRIPTION
**Pull request by beccadax**
These commits fix all broken `-t` tests on macOS; I'll be checking `-T` next, but I figure this is a good starting point.

(No idea if submitting a PR to your branch is the best way to get this stuff into your apple/swift PR, but I thought it was worth a try. If there's a better way to do it, let me know.)

### CODE REVIEW

**Comment by slavapestov** [comment]
Oh I didn't see this until now, sorry. Do you want to create a new PR and add my patch to it against Apple/swift instead? I'd want @rudkx to look at the changes too.

File: lib/Sema/CSGen.cpp [hunk 1]
```diff
@@ -1345,8 +1345,10 @@ namespace {
       // We may, alternatively, want to use a type variable in that case,
       // and possibly infer the type of the variable that way.
       CS.getTypeChecker().validateDecl(E->getDecl());
-      if (E->getDecl()->isInvalid())
+      if (E->getDecl()->isInvalid()) {
+        E->setType(E->getDecl()->getInterfaceType());
         return nullptr;
+      }
```
Comment (rudkx) [inline lib/Sema/CSGen.cpp]: I've been working on moving the constraint system over to use a side map for types, which are accessed via ConstraintSystem::getType(Expr *) and set via ConstraintSystem::setType(Expr *, Type), so those should be used here (at the moment they still set the type directly on the expression, but that will be going away).

**Review by rudkx — CHANGES_REQUESTED** [review]

File: lib/Sema/CSGen.cpp (hunk 1 above)
Comment (beccadax) [reply lib/Sema/CSGen.cpp]: Right, I've actually used that a couple times—sorry! Would the correct line be this?

    CS.setType(E, E->getDecl()->getInterfaceType());

You mention `ConstraintSystem::getType()`, but I don't think that works on `Decl`s.

**Comment by beccadax** [comment]
@slavapestov Once I've heard from @rudkx, I'll make a PR against apple/swift. Thanks for your help on this!

File: lib/Sema/CSGen.cpp (hunk 1 above)
Comment (rudkx) [reply lib/Sema/CSGen.cpp]: Yes, I just mentioned that for completeness regarding setting/getting types on Exprs. Your CS.setType call looks correct!

**Pull request closed without merging**